app/services/exporters/tad_data_request.rb

```diff
@@ -0,0 +1,94 @@
+class Exporters::TadDataRequest
+  def initialize(cohort:, schedules:, courses:, file:)
+    @cohort = cohort
+    @schedules = schedules
+    @courses = courses
+    @file = file
+    @csv = []
+    @csv << csv_headers
+  end
+
+  def call
+    generate_data
+    save_data_to_csv
+  end
+
+  def applications
+    course_id = @courses.pluck(:id)
+    schedule_id = @schedules.pluck(:id)
+
+    accepted_applications_ids = Declaration.billable.where(declaration_type: "started").where(cohort: @cohort).pluck(:application_id)
+    Application.where(id: accepted_applications_ids, schedule_id:, course_id:)
+  end
+
+private
+
+  def generate_data
+    applications.each do |application|
```

Comment:
not sure how many applications there are likely to be getting serialized but can make this do it in batches pretty easily?

```ruby
applications.find_each(batch_size: 500)
```